src/storage/mod.rs

```diff
@@ -1851,11 +1909,21 @@ impl<E: Engine, L: LockManager, F: KvFormat> Storage<E, L, F> {
         }
         let deadline = Self::get_deadline(&ctx);
         let cf = Self::rawkv_cf(&cf, self.api_version)?;
+        let provider = self.causal_ts_provider.clone();
         let engine = self.engine.clone();
+        let concurrency_manager = self.concurrency_manager.clone();
         self.sched_raw_command(CMD, async move {
             if let Err(e) = deadline.check() {
                 return callback(Err(Error::from(e)));
             }
+            let key_guard = Self::get_raw_key_guard(&provider, concurrency_manager).await;
+            if let Err(e) = key_guard {
+                return callback(Err(e));
+            }
+            let ts = Self::get_causal_ts(&provider);
+            if let Err(e) = ts {
+                return callback(Err(e));
+            }
             let command_duration = tikv_util::time::Instant::now();
```

Comment:
`let command_duration = tikv_util::time::Instant::now();` should be consistent with the position of in `raw_batch_put` (line 2026)